lib/api.js

```diff
@@ -0,0 +1,65 @@
+'use strict';
+
+// local modules
+
+const auth = require('./auth');
+const httpUtils = require('./utils/http');
+
+// this module
+
+const TYPES = [ 'answerspaces', 'interactions' ];
+
+function assertTypeAndId (options) {
+  if (!options.type || !options.id) {
+    throw new TypeError('"type" and "id" must be provided');
+  }
+}
+
+function assertKnownType (options) {
+  if (!~TYPES.indexOf(options.type)) {
+    throw new TypeError(`"${options.type}" is not a known type`);
+  }
+}
+
+function getDashboard () {
+  return auth.read()
+    .then((a) => httpUtils.sendRequest({
+      credential: a.credential,
+      url: `${a.origin}/_api/v1/dashboard`
+    }));
+}
+
+function getResource (options) {
+  options = options || {};
+  assertTypeAndId(options);
+  assertKnownType(options);
+
+  return auth.read()
+    .then((a) => httpUtils.sendRequest({
+      credential: a.credential,
+      url: `${a.origin}/_api/v1/${options.type}/${options.id}`
+    }));
+}
+
+function putResource (options) {
+  options = options || {};
+  assertTypeAndId(options);
```

Comment:
you could be super DRY here by making a wrapper function eg

    var wrapper = fn =>  options => { 
       /* common checking code here */ 
       return fn(options);
    }

then when you do exports wrap the functions that need the assertion